public/scripts/scripts.js

```diff
@@ -42,25 +40,26 @@ function getPlaylistTracks(access_token, allCallSongs, request_url, playListTrac
       console.log(tracks);
       tracks.map(function(track){
         playListTracks.push(track);
-        writePlayListToPanel(track, playListTracks);
       })
       console.log(playListTracks);
       if(response.next) {
         getPlaylistTracks(access_token, allCallSongs, response.next, playListTracks);
+      } else {
+        console.log('done making spotify api requests for this playlist');
+        initializePlayListControl(playListTracks);
+        writePlayListToPanel(playListTracks);
       }
     }
   });
-  $(document).ajaxComplete(function(){
-    console.log('all calls are complete, and now PLT is like so: ' + playListTracks);
-    initializePlayListControl(playListTracks);
-  });
-  return(playListTracks);
 }
 
-function writePlayListToPanel(track, playListTracks){
-  var thisParticulartrack = track;
-  var list = "<li id=\"" + thisParticulartrack.track.id + "\" class='playlistItem'>" + thisParticulartrack.track.name + "<br><span class=\"trackArtist\"> by " + thisParticulartrack.track.artists[0].name + "</span></li>"
-  document.getElementById('trackList').innerHTML += list;
+// TODO document me
+function writePlayListToPanel(playListTracks){
```

Comment:
this is the change i mentioned above - `writePlayListToPanel()` now takes a single `playListTracks` argument, and does its own `.map()` call so that its callers don't have to do that every time.

as a result, we only call `initTrackListener` a single time after a playlist is written to the sidebar; that's good, because i'm not entirely sure how it behaves if you call it a bunch of times while a playlist is in the middle of being written to the sidebar. you only need to call it once at the end, you don't need to call it once per track, since it operates on all elements that match the `'li.playlistItem'` selector.